### PR TITLE
Add custom error handler when fetching data

### PR DESCRIPTION
**Pull request by gregoriusjimmy**
## Type of changes

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Enhancement
- [x] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behaviour?

- There is no custom error handler

## What is the new behavior?

- Add custom error handler
- Refactor error handling in page/component to use new custom error handler
- Refactor responses
## Checklist

- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have filled the `[Unreleased]` [Changelog](https://github.com/nodefluxio/weber/blob/main/CHANGELOG.md

## Security Pentester API Checklist
- [ ] IDOR
- [ ] Malicious Script Injection
- [ ] SQL Injection

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by gregoriusjimmy** [comment]
@krismp tolong bantu review yak ini ada `error handling` baru

**Review by williammfu — APPROVED** [review]

File: frontend/app/api/paymentAPI.ts [hunk 1]
```diff
@@ -66,25 +49,9 @@ export const activateAccount = async (
       return {
         ...res.data
       }
-    } else {
-      throw new Error(res.data.message)
     }
   } catch (e) {
-    const axiosError = e as AxiosError<ActivationResponse>
-    console.log(axiosError.message)
-    if (axiosError.response?.status === 401) {
```
Comment (krismp) [inline frontend/app/api/paymentAPI.ts:75]: ini ada pengecekan status code juga @gregoriusjimmy udh di cek klo di handle errorHandler gmn?

File: frontend/app/components/modules/FaceEnrollment/FaceEnrollment.tsx [hunk 2]
```diff
@@ -66,15 +66,23 @@ export const FaceEnrollment = ({ openModal, payload, nextStep }: Props) => {
...
           setIsSuccess(true)
         } else {
-          if (res.message === SESSION_ID_ERROR) {
-            openModal()
-          }
+          throw new Error()
         }
       } catch (e) {
-        console.error((e as Error).message)
+        if (e instanceof CustomError) {
```
Comment (krismp) [inline frontend/app/components/modules/FaceEnrollment/FaceEnrollment.tsx:75]: e ini ada instance lain selain CustomError kah? jadi perlu di check gini @gregoriusjimmy ?

File: frontend/app/api/paymentAPI.ts (hunk 1 above)
Comment (gregoriusjimmy) [reply frontend/app/api/paymentAPI.ts:75]: @rizkybaihaqy yg skrng ini udah di handle di componentnya kan ki? jadi di handle di depannya

File: frontend/app/components/modules/FaceEnrollment/FaceEnrollment.tsx (hunk 2 above)
Comment (gregoriusjimmy) [reply frontend/app/components/modules/FaceEnrollment/FaceEnrollment.tsx:75]: gaada sih ka @krismp  cuman takut ada `Error` yang gaketanglep. Apa diapus aja?

Comment (krismp) [reply frontend/app/components/modules/FaceEnrollment/FaceEnrollment.tsx:75]: gw juga takut klo di apus :( wkwkwk

eh btw yg catch itu bisa gini `(catch: CustomError)` gak ya jim?

Comment (gregoriusjimmy) [reply frontend/app/components/modules/FaceEnrollment/FaceEnrollment.tsx:75]: belom pernah liat yg kyak gitu sih ka gw, https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch

Comment (krismp) [reply frontend/app/components/modules/FaceEnrollment/FaceEnrollment.tsx:75]: ok jim

**Review by rizkybaihaqy — APPROVED** [review]

File: frontend/app/api/paymentAPI.ts (hunk 1 above)
Comment (rizkybaihaqy) [reply frontend/app/api/paymentAPI.ts:75]: Iyak, dr fungsi errorHanldernya udh ngembalii error code. Jadi tinggal gimana di depannya lagi aja handle mau di apain.

Comment (gregoriusjimmy) [reply frontend/app/api/paymentAPI.ts:74]: aman yak? @krismp

Comment (krismp) [reply frontend/app/api/paymentAPI.ts:74]: gokil

**Review by krismp — APPROVED** [review]

**Pull request merged**